### ISSUE TITLE
[Bug]: NullReference when using `IPeripheral.ReadDeviceInformation()`

### ISSUE THREAD

**Issue by olicooper** [shinyorg/shiny#742]
### Component

BluetoothLE Client

### What platform(s) are effected?

- [ ] iOS 12
- [ ] iOS 13
- [ ] iOS 14
- [ ] Android 8
- [ ] Android 9
- [X] Android 10
- [ ] Android 11
- [ ] UWP

### Steps To Reproduce

I was calling it like this:

```csharp
IBleManager BleManager { get; }

//...

this.BleManager
    .GetConnectedPeripherals()
    .Subscribe(async (peripherals) =>
    {
        foreach (var peripheral in peripherals)
        {
            // Exception thrown here
            var info = await peripheral.ReadDeviceInformation();
        }
    });
```

### Expected Behavior

Device information should be returned

### Actual Behavior

App crashes with 'NullReference' error

### Exception or Log output

```log
**System.NullReferenceException:** 'Object reference not set to an instance of an object.'

[mono-rt] [ERROR] FATAL UNHANDLED EXCEPTION: System.NullReferenceException: Object reference not set to an instance of an object.
[mono-rt]   at Shiny.BluetoothLE.Peripheral.<GetServices>b__16_0 (System.IObserver`1[T] ob) [0x0008d] in /_/src/Shiny.BluetoothLE/Platforms/Android/Peripheral.cs:124 
[mono-rt]   at System.Reactive.Linq.QueryLanguage+CreateWithDisposableObservable`1[TSource].SubscribeCore (System.IObserver`1[T] observer) [0x00000] in /_/Rx.NET/Source/src/System.Reactive/Linq/QueryLanguage.Creation.cs:35 
[mono-rt]   at System.Reactive.ObservableBase`1[T].Subscribe (System.IObserver`1[T] observer) [0x00037] in /_/Rx.NET/Source/src/System.Reactive/ObservableBase.cs:58 
[mono-rt] --- End of stack trace from previous location where exception was thrown ---
[mono-rt] 
[mono-rt]   at Shiny.BluetoothLE.Internals.PeripheralContext+<>c__DisplayClass21_0`1[T].<Invoke>b__0 (System.Threading.CancellationToken ct) [0x000a7] in /_/src/Shiny.BluetoothLE/Platforms/Android/Internals/PeripheralContext.cs:76 
[mono-rt]   at System.Reactive.PlatformServices.ExceptionServicesImpl.Rethrow (System.Exception exception) [0x00000] in /_/Rx.NET/Source/src/System.Reactive/Internal/ExceptionServicesImpl.cs:19 
[mono-rt]   at System.Reactive.ExceptionHelpers.Throw (System.Exception exception) [0x00000] in /_/Rx.NET/Source/src/System.Reactive/Internal/ExceptionServices.cs:16 
[mono-rt]   at System.Reactive.Subjects.AsyncSubject`1[T].GetResult () [0x00038] in /_/Rx.NET/Source/src/System.Reactive/Subjects/AsyncSubject.cs:430 
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by aritchie** [comment]
3 things here
1) The issue isn't with ReadDeviceInformation at all.  It isn't able to retrieve a service because a GATT hasn't been established with your app.  The OS says that it is connected, but if you read the peripheral, it will be "disconnected" because you still have to do the connection.

2) Shiny isn't crashing your app, you have to trap the error in subscribe always.

3) You shouldn't blindly loop through GetConnectedPeripherals as it will return EVERYTHING, it doesn't mean you app is accessing the GATT on it, and in many cases, peripherals that don't have a device info gatt services..

**Comment by olicooper** [comment]
Thanks for your quick response! Sorry this turned out to be a support query rather than a bug report.. I have worked with .NET for a long time but only recently started mobile app and BLE development so these concepts are still new to me.

The comment for `IBleManager.GetConnectedPeripherals()` currently reads "Gets a list of connected peripherals _by your app_" so I assumed I was able to work with the devices straight away. In the connected device list I have a smart watch that must be getting connected by another app's background service - not shiny.

I think modifying the comment on `GetConnectedPeripherals()` would help clarify things i.e. "Gets a list of connected peripherals. You must ensure your app is connected to the IPeripheral before operating on it".

For me, I was trying to check if I was already connected to a specific device, and if not, start a scan to find it. I wanted to use the "Manufacturer Specific Information" to identify it by company/model, but that is only available in advertisement packets and advertisements are disabled once the device is connected. From what I gather, I will need to use the ServiceId's instead because the device Name and MAC can change.

Thanks for your help 👍
